lighthouse-core/report/html/renderer/performance-category-renderer.js

```diff
@@ -47,41 +47,27 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
    * @return {!Element}
    */
   _renderOpportunity(audit, index, scale) {
-    const tmpl = this.dom.cloneTemplate('#tmpl-lh-opportunity', this.templateContext);
-    const element = this.dom.find('.lh-load-opportunity', tmpl);
-    element.classList.add(`lh-load-opportunity--${Util.calculateRating(audit.result.score)}`);
+    const oppTmpl = this.dom.cloneTemplate('#tmpl-lh-opportunity', this.templateContext);
+    const element = this.renderAudit(audit, index, oppTmpl);
+    element.classList.add(`lh-audit--${Util.calculateRating(audit.result.score)}`);
     element.id = audit.result.name;
 
-    const titleEl = this.dom.find('.lh-load-opportunity__title', tmpl);
-    titleEl.textContent = audit.result.description;
-    this.dom.find('.lh-audit__index', element).textContent = `${index + 1}`;
-
-    if (audit.result.debugString || audit.result.error) {
-      const debugStrEl = this.dom.createChildOf(titleEl, 'div', 'lh-debug');
-      debugStrEl.textContent = audit.result.debugString || 'Audit error';
-    }
-    if (audit.result.error) return element;
-
     const details = audit.result.details;
     const summaryInfo = /** @type {!DetailsRenderer.OpportunitySummary}
     */ (details && details.summary);
     if (!summaryInfo || !summaryInfo.wastedMs) {
       return element;
     }
 
-    const displayValue = Util.formatDisplayValue(audit.result.displayValue);
+    const wastedEl = this.dom.find('.lh-audit__display-text', element);
     const sparklineWidthPct = `${summaryInfo.wastedMs / scale * 100}%`;
-    const wastedMs = Util.formatSeconds(summaryInfo.wastedMs, 0.01);
-    const auditDescription = this.dom.convertMarkdownLinkSnippets(audit.result.helpText);
-    this.dom.find('.lh-load-opportunity__sparkline', tmpl).title = displayValue;
-    this.dom.find('.lh-load-opportunity__wasted-stat', tmpl).title = displayValue;
-    this.dom.find('.lh-sparkline__bar', tmpl).style.width = sparklineWidthPct;
-    this.dom.find('.lh-load-opportunity__wasted-stat', tmpl).textContent = wastedMs;
-    this.dom.find('.lh-load-opportunity__description', tmpl).appendChild(auditDescription);
-
-    // If there's no `type`, then we only used details for `summary`
-    if (details.type) {
-      element.appendChild(this.detailsRenderer.render(details));
+    this.dom.find('.lh-sparkline__bar', element).style.width = sparklineWidthPct;
+    wastedEl.textContent = Util.formatSeconds(summaryInfo.wastedMs, 0.01);
+
+    if (audit.result.displayValue) {
```

Comment:
so if we're overriding displayValue anyhow and there's no helpText, we're using renderAudit for 

- populating index
- setting rating class
- sticking it inside of details
- filling error/debugString
- something else I'm missing?

WDYT about extracting those 4 to a method they both share?

Reply:
i like the sound of that. will do.